src/test/java/org/openrefine/extensions/commons/utils/CommonsImportingControllerTest.java

```diff
@@ -198,4 +199,48 @@ public void testgetNextRowOfCells() throws Exception {
         }
     }
 
+    /**
+     * Test row generation from multiple api calls and paging with a cmcontinue token
+     */
+    @Test
+    public void testgetNextRowOfCellsMultCategories() throws Exception {
+
+        try (MockWebServer server = new MockWebServer()) {
+            server.start();
+            HttpUrl url = server.url("/w/api.php");
+            String jsonResponseCategory1 = "{\"batchcomplete\":\"\",\"query\":{\"categorymembers\":[{\"pageid\":187322,\"ns\":3,"
+                    + "\"title\":\"File:The Letter A.jpg\",\"type\":\"file\"}]}}";
+            server.enqueue(new MockResponse().setBody(jsonResponseCategory1));
+            String jsonResponseCategory2 = "{\"batchcomplete\":\"\",\"continue\":{\"cmcontinue\":\"file|4492e4a5047|13935\",\"continue\":\"-||\"},"
+                    + "\"query\":{\"categorymembers\":[{\"pageid\":127722,\"ns\":6,"
+                    + "\"title\":\"File:The Letter B.jpg\",\"type\":\"file\"}]}}";
+            server.enqueue(new MockResponse().setBody(jsonResponseCategory2));
+            String jsonResponseCategory2Continue = "{\"batchcomplete\":\"\",\"query\":{\"categorymembers\":[{\"pageid\":112928,\"ns\":6,"
+                    + "\"title\":\"File:La letra B.jpg\",\"type\":\"file\"}]}}";
+            server.enqueue(new MockResponse().setBody(jsonResponseCategory2Continue));
+            /* create job.mock */
+            job = Mockito.mock(ImportingJob.class);
+            List<String> categories = new ArrayList<>();
+            categories.add("Category:A");
+            categories.add("Category:B");
+            FilesBatchRowReader reader = new FilesBatchRowReader(job, categories, url.toString());
+
+            List<Object> currentRow = null;
+            List<List<Object>> rows = new ArrayList<>();
+            while ((currentRow = reader.getNextRowOfCells()) != null) {
+                rows.add(currentRow);
+            }
+
+            RecordedRequest recordedRequestCategory1 = server.takeRequest();
+            String queryParameters = "/w/api.php?action=query&list=categorymembers&"
+                    + "cmtitle=Category%3AA&cmtype=file&cmprop=title%7Ctype%7Cids&cmlimit=500&format=json";
```

Comment:
Very nice! Perhaps you could do it for the other two requests: calling `server.takeRequest()` again will give you the second request, which you can run a similar assert on, and so on.